sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -64,6 +65,19 @@ class SparkSession private(
    |  Session-related state  |
    * ----------------------- */
 
+  {
+    val defaultWarehousePath =
+      SQLConf.WAREHOUSE_PATH
+        .defaultValueString
+        .replace("${system:user.dir}", System.getProperty("user.dir"))
+    val warehousePath = sparkContext.conf.get(
+      SQLConf.WAREHOUSE_PATH.key,
+      defaultWarehousePath)
+    sparkContext.conf.set(SQLConf.WAREHOUSE_PATH.key, warehousePath)
+    sparkContext.conf.set("hive.metastore.warehouse.dir", warehousePath)
```

Comment:
Currently, the `Set` command does not work if the property is `hive.x.y.z`. Will try to submit a PR for resolving that tonight. Thanks!